crates/ty_python_semantic/resources/mdtest/bidirectional.md

```diff
@@ -200,6 +200,41 @@ def f() -> list[Literal[1]]:
     return [1]
 ```
 
+## Instance attribute
+
+```toml
+[environment]
+python-version = "3.12"
+```
+
+Both meta and class/instance attribute annotations are used as type context:
+
+```py
+from typing import Literal, Any
+
+class DataDescriptor:
+    def __get__(self, instance: object, owner: type | None = None) -> list[Literal[1]]:
+        return []
+
+    def __set__(self, instance: object, value: list[Literal[1]]) -> None:
+        pass
+
+def lst[T](x: T) -> list[T]:
+    return [x]
+
+def _(flag: bool):
+    class Meta(type):
+        if flag:
+            x: DataDescriptor = DataDescriptor()
```

Comment:
Is it necessary to partially shadow `x` on `C` here, or could this just be an attribute with a different name? (and a separate assignment test below)?